driver/pad_reflection_driver.hpp

```diff
@@ -0,0 +1,617 @@
+    std::unique_ptr<GPUMem> out_dev;
+
+    std::vector<Tgpu> in;
+    std::vector<Tgpu> out;
+    std::vector<Tref> outhost;
+    std::vector<Tref> inhost;
+
+    std::vector<size_t> padding;
+};
+
+template <typename Tgpu, typename Tref>
+int PadReflectionDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename T>
+inline std::vector<T> GetStrides(std::vector<T> input, bool contiguous)
+{
+    if(!contiguous)
+        std::swap(input.front(), input.back());
+    std::vector<T> strides(input.size());
+    strides.back() = 1;
+    for(int i = input.size() - 2; i >= 0; --i)
+        strides[i] = strides[i + 1] * input[i + 1];
+    if(!contiguous)
+        std::swap(strides.front(), strides.back());
+    return strides;
+}
+
+template <typename Tgpu, typename Tref>
+int PadReflectionDriver<Tgpu, Tref>::GetandSetData()
+{
+    std::string padding_str = inflags.GetValueStr("padding");
+    std::stringstream padding_ss(padding_str);
+    std::string padding_token;
+    while(std::getline(padding_ss, padding_token, ','))
+    {
+        padding.push_back(std::stoul(padding_token));
+    }
+
+    // if(!(padding.size() == 1 || padding.size() == 4))
+    if(!(padding.size() == 1))
+    {
+        std::cerr << "Error Padding Lengths\n" << std::endl;
+    }
+    std::vector<int> in_len = GetInputTensorLengthsFromCmdLine();
+
+    auto input_strides = GetStrides(in_len, contiguous == 1);
+    SetTensorNd(inputDesc, in_len, input_strides, data_type);
+
+    std::vector<int> out_len;
+    auto in_len_size = in_len.size();
+    if(in_len_size == 3)
+    {
+        for(int i = 0; i < in_len_size; i++)
+        {
+            // If W
+            if(i == 2)
+            {
+                out_len.push_back(in_len[i] + 2 * padding[0]);
+            }
+            else
+            {
+                out_len.push_back(in_len[i]);
+            }
+        }
+    }
+    else if(in_len_size == 4)
+    {
+        for(int i = 0; i < in_len.size(); i++)
+        {
+            // If H
+            if(i == 2)
+            {
+                out_len.push_back(in_len[i] + 2 * padding[2]);
+            }
+            // If W
+            else if(i == 3)
+            {
+                out_len.push_back(in_len[i] + 2 * padding[0]);
+            }
+            else
+            {
+                out_len.push_back(in_len[i]);
+            }
+        }
+    }
+
+    if(out_len.empty())
+        out_len.push_back(1);
+    auto output_strides = GetStrides(out_len, contiguous == 1);
+
+    SetTensorNd(outputDesc, out_len, output_strides, data_type);
+
+    return 0;
+}
+
+template <typename Tgpu, typename Tref>
+int PadReflectionDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward Pad Reflection (Default=1)", "int");
```

Comment:
Update the default values or the description.

Reply:
resolve in comit [2fd809d](https://github.com/moreh-dev/MIOpen/pull/15/commits/2fd809d8489d7fd1880e54d72ffd94f664f3ee4e)